ratchet/ratchet.go

```diff
@@ -97,37 +91,42 @@ func (r *Ratchet) randBytes(buf []byte) {
 }
 
 func New(rand io.Reader) *Ratchet {
-	r := &Ratchet{
-		rand:       rand,
-		kxPrivate0: new([32]byte),
-		kxPrivate1: new([32]byte),
-		saved:      make(map[[32]byte]map[uint32]savedKey),
-	}
-
-	r.randBytes(r.kxPrivate0[:])
-	r.randBytes(r.kxPrivate1[:])
-
+	r := new(Ratchet)
+	r.rand = rand
+	r.kxPrivate = new([32]byte)
+	r.randBytes(r.kxPrivate[:])
+	r.saved = make(map[[32]byte]map[uint32]savedKey)
 	return r
 }
 
 type KeyExchange struct {
-	Dh  []byte
-	Dh1 []byte
+	Cipher [sntrup4591761.CiphertextSize]byte
+	Public []byte
 }
 
 // FillKeyExchange sets elements of kx with key exchange information from the
 // ratchet.
 func (r *Ratchet) FillKeyExchange(kx *KeyExchange) error {
-	if r.kxPrivate0 == nil || r.kxPrivate1 == nil {
-		return errors.New("ratchet: handshake already complete")
+	c, k, err := sntrup4591761.Encapsulate(r.rand, r.TheirPublicKey)
+	if err != nil {
+		return err
 	}
+	pub := new([32]byte)
+	curve25519.ScalarBaseMult(pub, r.kxPrivate)
+	blobshare.SetNrp(32768, 16, 2)
```

Comment:
Don't we have consts for these?

Reply:
Hi Marco,

Unfortunately we don't. This pattern is repeated elsewhere in our tree:

./zkclient/kxwindow.go:		blobshare.SetNrp(32768, 16, 2)
./zkclient/kxacceptwindow.go:		blobshare.SetNrp(32768, 16, 2)
./ratchet/ratchet.go:	blobshare.SetNrp(32768, 16, 2)
./ratchet/ratchet.go:	blobshare.SetNrp(32768, 16, 2)
./ratchet/ratchet_test.go:	blobshare.SetNrp(32768, 16, 2)